src/vm/script.rs

```diff
@@ -3659,6 +3659,105 @@ impl<'a> ScriptExecutor<'a> {
                     }
                 },
 
+                ScriptEntry::Opcode(OP::Blake3_256Keyed) => {
+                    if exec_stack.len() < 2 {
+                        self.state = ScriptExecutorState::Error(
+                            ExecutionResult::InvalidArgs,
+                            (i_ptr, func_idx, op.clone(), exec_stack.as_slice()).into(),
+                        );
+                        return;
+                    }
+
+                    let value_to_hash = exec_stack.pop().unwrap();
+                    *memory_size -= value_to_hash.size();
+                    let new_key = exec_stack.pop().unwrap();
+                    *memory_size -= new_key.size();
+
+                    if new_key.get_type() != crate::vm::internal::HASH_KEY_TYPE {
+                        self.state = ScriptExecutorState::Error(
+                            ExecutionResult::InvalidArgs,
+                            (i_ptr, func_idx, op.clone(), exec_stack.as_slice()).into(),
+                        );
+                        return;
+                    }
+
+                    if let VmTerm::Unsigned8Array(arr) = new_key {
+                        let hash_term =
+                            bifs::blake3_256_internal(&value_to_hash, &hex::encode(arr));
```

Comment:
This shouldn't be encoded as hex but passed as a byte array. It should panic if the key is not valid utf8.

Reply:
We should use this crate for all utf8 operations: https://github.com/rusticstuff/simdutf8

Reply:
I thought about it and this should accept any kind of byte array as key. The problem is the internal hash functions take `&str` params and cast them to `&[u8]`. We should expose the key parameter as `&[u8]` instead.